app/src/main/java/com/hedvig/app/feature/home/model/HomeItemsBuilder.kt

```diff
@@ -0,0 +1,163 @@
+package com.hedvig.app.feature.home.model
+
+import androidx.compose.ui.unit.dp
+import arrow.core.NonEmptyList
+import com.hedvig.android.owldroid.graphql.HomeQuery
+import com.hedvig.android.owldroid.type.PayinMethodStatus
+import com.hedvig.app.R
+import com.hedvig.app.feature.claims.ui.commonclaim.CommonClaimsData
+import com.hedvig.app.feature.claims.ui.commonclaim.EmergencyData
+import com.hedvig.app.feature.home.ui.claimstatus.data.ClaimStatusCardUiState
+import com.hedvig.app.util.featureflags.Feature
+import com.hedvig.app.util.featureflags.FeatureManager
+
+@OptIn(ExperimentalStdlibApi::class)
+class HomeItemsBuilder(
+    private val featureManager: FeatureManager
+) {
+
+    fun buildItems(
+        homeData: HomeQuery.Data
+    ): List<HomeModel> = when {
+        homeData.isActive() -> buildActiveItems(homeData)
+        (homeData.isPending() || homeData.isActiveInFuture()) && homeData.isSwitching() -> buildSwitchingItems(homeData)
+        homeData.isPending() -> buildPendingItems(homeData)
+        homeData.isActiveInFuture() -> buildActiveInFutureItems(homeData)
+        homeData.isTerminated() -> buildTerminatedItems(homeData)
+        else -> listOf(HomeModel.Error)
+    }
```

Comment:
This way we have to make sure that since those are mutually exclusive, and only one of them runs that we have to specify on each case all the items that it needs, and not for example having this one long method that simply adds something to the long existing list. Feels like a win in the long run, where we need to be aware that when we make changes, we take care of all these mutually exclusive cases!